### PR TITLE
Orbax checkpointing

### PR DESCRIPTION
**Pull request by entrpn**
- Enable orbax checkpointing.
- Remove transformers library dependency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,7 +75,7 @@ After installation completes, run the training script.
 
   ```bash
   export LIBTPU_INIT_ARGS=""
-  python -m src.maxdiffusion.train_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_xl_run" base_output_directory="gs://your-bucket/" per_device_batch_size=1
+  python -m src.maxdiffusion.train_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_xl_run" output_dir="gs://your-bucket/" per_device_batch_size=1
   ```
```
Comment (suexu1025) [inline README.md:79]: question: are we adding extra output_dir? or just update the name as base_output_directory or output_dir?

File: src/maxdiffusion/max_utils.py [hunk 2]
```diff
@@ -342,98 +358,69 @@ def setup_initial_state(model, tx, config, mesh, model_params, unboxed_abstract_
...
-    init_train_state_partial = functools.partial(init_train_state, model=model, tx=tx, training=training)
-
-    sharding = PositionalSharding(mesh.devices).replicate()
-    partial_device_put_replicated = functools.partial(device_put_replicated, sharding=sharding)
-    model_params = jax.tree_util.tree_map(partial_device_put_replicated, model_params)
-
-    with jax.transfer_guard("disallow"):
+      state = checkpointing_utils.load_state_if_possible(checkpoint_manager,
+                                                  unboxed_abstract_state, checkpoint_item)
+      if state:
```
Comment (suexu1025) [inline src/maxdiffusion/max_utils.py:370]: Do we need to add check for checkpoint_item in case if it's none

File: src/maxdiffusion/max_utils.py [hunk 3]
```diff
@@ -342,98 +358,69 @@ def setup_initial_state(model, tx, config, mesh, model_params, unboxed_abstract_
...
+        state = state[checkpoint_item]
+    if not state:
+      max_logging.log(f"Could not find {checkpoint_item} in orbax, creating state...")
+      # for DDP needs replication before jit state.
+      if config.ici_data_parallelism == -1 or config.dcn_data_parallelism == -1:
```
Comment (suexu1025) [inline src/maxdiffusion/max_utils.py:375]: Is it  safe to add ici_data_parallelism == np.prod(all_devices_axis)?  we might do fsdp but still pass ici_data_parallelism  = -1

**Review by suexu1025 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (entrpn) [reply README.md:79]: its updating the name to output_dir.

**Pull request merged**